go/vt/vtgate/tabletgateway.go

```diff
@@ -301,10 +301,16 @@ func (gw *TabletGateway) withRetry(ctx context.Context, target *querypb.Target,
 					err = vterrors.Errorf(vtrpcpb.Code_CLUSTER_EVENT, "current keyspace is being resharded")
 					continue
 				}
-				if kev.PrimaryIsNotServing(target) {
+				primary, serving := kev.PrimaryIsNotServing(target)
+				if serving {
```

Comment:
nit: calling the return variable `serving` is misleading. This code block is hard to understand because if the value of the variable `serving` is true, we return an error that says "primary is not serving".

Reply:
Will be fixed via https://github.com/vitessio/vitess/pull/13106